docs/reference/environment-variables.rst

```diff
@@ -0,0 +1,101 @@
+*********************
+Environment variables
+*********************
+
+Applications built on craft-application have several environment variables that
+can configure their behaviour. They and the behaviour they modify are listed
+below.
+
+Variables passed to managed builders
+------------------------------------
+
+Several environment variables from the host environment are passed to the
+managed build environment. While an application may adjust these by adjusting
+the ``environment`` dictionary attached to the ``ProviderService``,
+craft-application will by default forward the ``http_proxy``, ``https_proxy``
+and ``no_proxy`` environment variables from the host.
+
+Supported variables
+-------------------
+
+These variables are explicitly supported for user configuration.
+
+.. _env-var-craft-build-environment:
+
+``CRAFT_BUILD_ENVIRONMENT``
+===========================
+
+If the value is ``host``, allows an environment to tell a craft application to run
+directly on the host rather than in managed mode. This method is roughly equivalent
+to using ``--destructive-mode``, but is designed for configurations where the
+application is already being run in an appropriate container or VM, such as
+`Snapcraft rocks <https://github.com/canonical/snapcraft-rocks/>`_ or
+when controlled by a CI system such as `Launchpad <https://launchpad.net>`_.
+
+**CAUTION**: Setting the build environment is only recommended if you are aware of
+the exact packages needed to reproduce the build containers created by the app.
+
+``CRAFT_BUILD_FOR``
+===================
+
+Sets the default architecture to build for. Overridden by ``--build-for`` in
+lifecycle commands.
+
+``CRAFT_PLATFORM``
+==================
+
+Sets the default platform to build. Is overridden by ``--platform`` in lifecycle
+commands.
+
+``CRAFT_SNAP_CHANNEL``
+======================
+
+Overrides the default channel that a craft application's snap is installed from if
+the manager instance is not running as a snap. If unset, the application will
+be installed from the ``latest/stable`` channel. If the application is running from
+a snap, this variable is ignored and the same snap used on the host system is
+injected into the managed builder.
+
+Development variables
+---------------------
+
+The following variables exist to help developers writing applications using
+craft-application more easily debug their code:
+
+``CRAFT_DEBUG``
+===============
+
+Controls whether the application is in debug mode. If this variable is set to
+``1``, general exceptions will not be caught, instead showing a traceback on
```

Comment:
possibly add a note saying that the traceback is always printed to the log file

Reply:
Good idea. Done!